lib/rubocop/cop/rails/duplicate_association.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop looks for associations that have been defined multiple times in the same file.
+      #
+      # @example
+      #
+      #   # bad
+      #   belongs_to :foo
+      #   belongs_to :bar
+      #   has_one :foo
+      #
+      #   # good
+      #   belongs_to :foo
+      #   has_one :bar
+      #
+      class DuplicateAssociation < Base
+        include RangeHelp
+        extend AutoCorrector
+
+        MSG = "Association `%<name>s` is defined multiple times. Don't repeat associations in a model."
+
+        def_node_matcher :association, <<~PATTERN
+          (send nil? ${:belongs_to :has_one :has_many :has_and_belongs_to_many} ({sym str} $_) ...)
+        PATTERN
+
+        def on_class(class_node)
+          offenses(class_node).each do |name, nodes|
+            length = nodes.length
+
+            nodes.each.with_index do |node, i|
+              add_offense(node, message: format(MSG, name: name)) do |corrector|
+                corrector.remove(range_by_whole_lines(node.source_range, include_final_newline: true)) if i < length - 1
+              end
+            end
+          end
+        end
+
+        private
+
+        def offenses(class_node)
+          class_send_nodes(class_node).select { |node| association(node) }
+                                      .group_by { |assoc| association_name(assoc) }
+                                      .select { |_, nodes| nodes.length > 1 }
+        end
+
+        def association_name(node)
+          node.first_argument.value.to_sym
+        end
+
+        def class_send_nodes(class_node)
+          class_def = class_node.body
+
+          return [] unless class_def
+
+          if class_def.send_type?
+            [class_def]
+          else
+            class_def.each_child_node(:send).to_a
```

Comment:
Is `to_a` necessary? I think `each_child_node` returns an iterator.

Reply:
It's not. Should I remove it from [after_commit_override](https://github.com/rubocop/rubocop-rails/blob/master/lib/rubocop/cop/rails/after_commit_override.rb#L74) too then? That's where I copied it from.

Or, should I move this shared code somewhere else to DRY it? It seems really helpful for any time you want to capture a bunch of class method calls in `on_class`.

Reply:
Fair enough. 👍 
Yes, `to_a` looks redundant there, too.
Please have no hesitation to extract `class_send_nodes` into a mixin if you're up for it.

Reply:
I could do that! Would that be a follow up PR or part of this one? Or perhaps I do it in the duplicate scopes PR?

Reply:
The duplicate scopes PR seems to be a reasonable choice 👍

Reply:
Sounds good to me! I'll do that after this one gets merged.